manager/state/raft/raft.go

```diff
@@ -228,7 +245,7 @@ func (n *Node) WithContext(ctx context.Context) (context.Context, context.Cancel
 }
 
 // JoinAndStart joins and starts the raft server
-func (n *Node) JoinAndStart(ctx context.Context) (err error) {
+func (n *Node) JoinAndStart(ctx context.Context, keyRotator EncryptionKeyRotator) (err error) {
```

Comment:
I think this new parameter would fit better on `NewNode`.  Are there any issues with moving it there?